### PR TITLE
add cache key object to support caching with metadata

### PR DESCRIPTION
**Pull request by nwadams**
@lxdvs 

Add support for cache key metadata so we can have more efficient image retrieval.

### CODE REVIEW

File: library/src/main/java/com/android/volley/toolbox/CacheKey.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.android.volley.toolbox;
+
+public class CacheKey {
```
Comment (elihart) [inline library/src/main/java/com/android/volley/toolbox/CacheKey.java]: Seems like this name should reflect that this is a cache key to be used with images. It doesn't make sense to have a width and height for other cache keys right?

Comment (felipecsl) [inline library/src/main/java/com/android/volley/toolbox/CacheKey.java]: agree with @elihart. Make it maybe `ImageCacheKey` and move the class member to the top

**Comment by nwadams** [comment]
renamed and moved.

**Comment by felipecsl** [comment]
:+1:

**Pull request merged**